### PR TITLE
Refactor: Admin Resolver Events and Logging

### LINKED ISSUE #2194: 🔧 [Refactor] Admin Resolver Events and Logging
## 🔧 Refactor ticket

Check the queries and mutations of the admin resolver.
Ensure that:
 * All errors are logged with logger.error before throwing
 * All events defined are saved in the DB
 * Add unit tests for the logger (expect to be called ...) and events (check that the event is in the DB with the correct data)
 * See if more log messages could be helpful

### PR DESCRIPTION
**Pull request by jjimenezgarcia**
## 🍰 Pullrequest
Add all missing events, logs, and corresponding tests to admin resolver.

### Issues

- fixes #2194 

### Todo

- [x] Add logs
- [x] Add tests
- [x] Add missing events

### CODE REVIEW

**Review by clauspeterhuebner — DISMISSED** [review]
Good work!
Only one simple question as comment...

File: backend/src/graphql/resolver/AdminResolver.test.ts
Comment (Mogge) [reply backend/src/graphql/resolver/AdminResolver.test.ts:1028]:
```javascript
new Date('not-valid')
 - > Invalid Date

new Date('not-valid').toString()
 - > "Invalid Date" 
```

**Review by Mogge — CHANGES_REQUESTED** [review]
Please add more events:
`DELETE_USER`
`CHANGE_USER_ROLE`
`UNDELETE_USER`
`ADMIN_UPDATE_CONTRIBUTION`
`ADMIN_DELETE_CONTRIBUTION`
`CREATE_CONTRIBUTION_LINK`
`DELETE_CONTRIBUTION_LINK`
`UPDATE_CONTRIBUTION_LINK`
`ADMIN_CREATE_CONTRIBUTION_MESSAGE`

File: backend/src/graphql/resolver/AdminResolver.test.ts
Comment (jjimenezgarcia) [reply backend/src/graphql/resolver/AdminResolver.test.ts:1028]: solved

**Review by Mogge — COMMENTED** [review]
I wait for the review until the other branch is merged

**Comment by Mogge** [comment]
@jjimenezgarcia Please resolve this conflict

**Comment by Mogge** [comment]
@jjimenezgarcia Ping

File: backend/src/graphql/resolver/AdminResolver.test.ts
Comment (Mogge) [inline backend/src/graphql/resolver/AdminResolver.test.ts:1362]: Please have this skipped. We do not want to have the month changed for now.

**Review by Mogge — CHANGES_REQUESTED** [review]
Very good! Just one small thing. As you can see, the tests are a bit flaky. Furthermore, changing the month on update contribution is not behaving as expected. Therefore we switched of the option to change the month. Please have the tests skipped.

File: backend/src/graphql/resolver/AdminResolver.test.ts
Comment (jjimenezgarcia) [reply backend/src/graphql/resolver/AdminResolver.test.ts:1362]: fixed

**Review by Mogge — APPROVED** [review]

**Pull request merged**
